src/Storages/MergeTree/MergeTreeDataMergerMutator.cpp

```diff
@@ -1941,21 +1944,20 @@ std::set<MergeTreeIndexPtr> MergeTreeDataMergerMutator::getIndicesToRecalculate(
     return indices_to_recalc;
 }
 
-std::set<MergeTreeProjectionPtr> MergeTreeDataMergerMutator::getProjectionsToRecalculate(
+std::set<ProjectionDescriptionRawPtr> MergeTreeDataMergerMutator::getProjectionsToRecalculate(
```

Comment:
BTW, why we use `std::set` of raw pointers here? Order of pointers is important?

Reply:
There should be negligible difference and IIUC for < 10 elements red-black tree performs better than hash tables.